packages/cli/lib/utils.js

```diff
@@ -224,13 +224,21 @@ exports.findArtifactPaths = async function(dir, artifactType, reader) {
   const readdir = reader || readdirAsync;
   debug(`Finding artifact paths at: ${dir}`);
 
-  // Wrapping readdir in case it's not a promise.
-  const files = await readdir(dir);
-  return _.filter(files, f => {
-    return (
-      _.endsWith(f, `${artifactType}.js`) || _.endsWith(f, `${artifactType}.ts`)
+  try {
+    // Wrapping readdir in case it's not a promise.
+    const files = await readdir(dir);
+    return files.filter(
+      f =>
+        _.endsWith(f, `${artifactType}.js`) ||
+        _.endsWith(f, `${artifactType}.ts`),
     );
-  });
+  } catch (err) {
+    if (err.code === 'ENOENT') {
```

Comment:
I would prefer to check if the directory exists in advance instead of using the err code.

Reply:
I find that approach problematic, because it introduces a race condition - between the time we check if the directory exists, and the time we read the entries, the directory can be deleted or removed.

What are your concerns about my proposed approach?

Reply:
I don't think we need to deal with the edge case where the directory is removed when the command in running. If so, we'll have to use sync fs apis.

My main concern is to rely on the error code. It would be more authoritative by calling an api to check existence of the directory.

It's not a blocker for the PR, anyway.